serpentTools/parsers/microxs.py

```diff
@@ -0,0 +1,271 @@
+            for tline in fid:
+                if 'NFY' in tline:
+                    return
+            raise SerpentToolsException("Fission yields values were not "
+                                        "found in {}".format(self.filePath))
+
+    def _postcheck(self):
+        """ensure the parser grabbed expected results."""
+        if not self.xsVal and not self.fluxRatio and not self.nfy:
+            raise SerpentToolsException("xs, fluxes and fission yields are all"
+                                        " empty in {}".format(self.filePath))
+
+    def getFY(self, parent, energy, daughter, flagEnergy=True):
+        """
+        Return a specific fission yield given the parent ID, neutron energy
+         and daughter ID
+
+        If the energy does not exist in the results, the fission yield
+        corresponding to the closest energy is returned
+
+        Parameters
+        ----------
+        parent: int or float
+            ID of the parent undergoing fission
+        energy: float
+            neutron energy in MeV
+        daughter: int or float
+            ID of the fission product
+        flagEnergy: boolean
+            If set to true, the function will return the fission yield
+            that matches to the closest energy given by the user
+
+        Returns
+        -------
+        Independent and cumulative fission yields: float or empty
+            IndYield and CumYield
+            if fission product exists, float values are returned,
+            otherwise empty values are returned
+
+        Raises
+        ------
+        SerpentToolsException:
+            If energy is a negative number
+            If parent or fission product are not found
+        """
+        IndYield, CumYield = 0, 0
+        if energy < 0:
+            raise SerpentToolsException("Energy entry {0} must be positive"
+                                        " in {1}".format(energy, self.filePath))
+        eneList = []
+        if flagEnergy:
+            # obtain the different energies for a specific parent
+            eneList = [items[1] for items in self.nfy.keys() if parent == items[0]]
+            # obtain the closest energy value
+            energy = min(eneList,
+                         key=lambda x: abs(x - energy)) if eneList else energy
+        if (parent, energy) not in self.nfy.keys():
+            raise SerpentToolsException("There is no parent {0} with energy "
+                                        "{1} in {2}".format(parent,energy,
+                                                            self.filePath))
+
+        FP = self.nfy[(parent, energy)]['fissProd']
+        IndYield = self.nfy[(parent, energy)]['indYield']
+        CumYield = self.nfy[(parent, energy)]['cumYield']
+        if daughter:
+            IndYield = IndYield[FP == daughter]
+            CumYield = CumYield[FP == daughter]
+        return IndYield, CumYield
+
+
+    def getXS(self, universe, isotope, reaction, isomeric=0):
```

Comment:
Should we also store the materials inside the universe? You can specify multiple materials and multiple universes from the setting, so how do we differentiate?

Reply:
I have just ran a case with different universes and materials just to make sure. 
You can indeed defined multiple materials, however the cross-sections presented would represent the isotopes enclosed in this universe. In other words, the output would contain:

% Universe: 0
% materials: fuel w70p05

FLUX_RATIO_0 = [  8.85236E-01 0.00018  8.85092E-01 0.00036 ];

XS_0 = [
922350   18  0  2.73679E+00 0.00162  7.87025E+01 0.00124
541350  102  0  0.00000E+00 0.00000  0.00000E+00 0.00000
 10010  102  0  2.04915E-03 0.00103  9.78808E-02 0.00134
];

Reply:
In addition, Serpent does not allow to set mdep for the same universe. 

This one is allowed:
set gcu 1 0

set mdep 0 1 2 w70p05 fuel
922350 18
541350 102
10010  102

set mdep 1 1 1 fuel
922350 18
541350 102
10010  102

This one is NOT allowed:

set gcu 1 0

set mdep 0 1 1 w70p05 
922350 18
541350 102
10010  102

set mdep 0 1 1 fuel
922350 18
541350 102
10010  102

Reply:
Okay. So for the case where you have `w70p05` and `fuel` in the `mdep` setting, how are they portrayed in the output file? What will happen if the current implementation is used to read this file?

Reply:
The fuel material will not be used as an indicator in the output, just the universe name. 

Here, how the output looks like in Serpent. The universe '0' and '1' are used in the variable names: 'XS_' and 'FLUX_RATIO_'

% Universe: 0
% materials: fuel w70p05

FLUX_RATIO_0 = [  8.85236E-01 0.00018  8.85092E-01 0.00036 ];

XS_0 = [
922350   18  0  2.73679E+00 0.00162  7.87025E+01 0.00124
541350  102  0  0.00000E+00 0.00000  0.00000E+00 0.00000
 10010  102  0  2.04915E-03 0.00103  9.78808E-02 0.00134
];

% Universe: 1
% materials: fuel

FLUX_RATIO_1 = [  3.32657E-01 0.00031  3.08654E-01 0.00040 ];

XS_1 = [
922350   18  0  7.28290E+00 0.00150  2.25687E+02 0.00136
541350  102  0  0.00000E+00 0.00000  0.00000E+00 0.00000
 10010  102  0  0.00000E+00 0.00000  0.00000E+00 0.00000
];